pkg/daemon/controller/controller.go

```diff
@@ -0,0 +1,311 @@
+	err = clientset.List(context.TODO(), &nodes)
+	if err != nil {
+		return err
+	}
+	addrs, err := localInterfaceAddresses()
+	if err != nil {
+		return err
+	}
+	node, err := currentNode(&nodes, addrs)
+	if err != nil {
+		return err
+	}
+
+	desiredVersion, present := node.Annotations["windowsmachineconfig.openshift.io/desired-version"]
+	if !present {
+		return errors.New("node missing desired version annotation")
+	}
+	var cm core.ConfigMap
+	err = clientset.Get(context.TODO(), client.ObjectKey{
+		Namespace: "openshift-windows-machine-config-operator", Name: servicescm.NamePrefix + desiredVersion}, &cm)
+	if err != nil {
+		return err
+	}
+	data, err := servicescm.Parse(cm.Data)
+	if err != nil {
+		return err
+	}
+
+	c := NewServiceController(clientset, svcMgr, node.Name)
+	// TODO: Instead of reconciling services only once, we need to watch managed Windows services for changes and
+	//       reconcile services if a change happens, as well as reconciling if any changes occur to the ConfigMap
+	//       or node object.
+	return c.reconcileServices(data.Services)
+}
+
+// localInterfaceAddresses returns a slice of all addresses associated with local network interfaces
+func localInterfaceAddresses() ([]net.Addr, error) {
+	var addresses []net.Addr
+	netIfs, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+	for _, netInterface := range netIfs {
+		addrs, err := netInterface.Addrs()
+		if err != nil {
+			return nil, err
+		}
+		addresses = append(addresses, addrs...)
+	}
+	return addresses, nil
+}
+
+// currentNode returns the node associated with this VM
+func currentNode(nodes *core.NodeList, localAddrs []net.Addr) (*core.Node, error) {
+	for _, localAddr := range localAddrs {
+		ipAddr, ok := localAddr.(*net.IPNet)
+		if !ok {
+			continue
+		}
+		ipv4Addr := ipAddr.IP.To4()
+		if ipv4Addr == nil || ipv4Addr.IsLoopback() {
+			continue
+		}
+		// Go through each node and check if the node has the ipv4 address in the address slice
+		if node := nodeutil.FindByAddress(ipv4Addr.String(), nodes); node != nil {
+			return node, nil
+		}
+	}
+	return nil, errors.New("unable to find associated node")
+}
+
+// NewServiceController returns a pointer to a ServiceController object
+func NewServiceController(client client.Client, mgr winsvc.Mgr, nodeName string) *ServiceController {
+	return &ServiceController{client: client, svcMgr: mgr, ctx: context.TODO(), nodeName: nodeName}
+}
+
+// existingServices returns a map with the keys being all Windows services currently present on the VM
+func (c *ServiceController) existingServices() (map[string]struct{}, error) {
+	// The most reliable way to determine if a service exists or not is to do a 'list' API call. It is possible to
+	// remove this call, and parse the error messages of a service 'open' API call, but I find that relying on human
+	// readable errors could cause issues when providing compatibility across different versions of Windows.
+	svcList, err := c.svcMgr.ListServices()
```

Comment:
nit: have you considered not giving a name to svcMgr in the struct so we can directly access all of it's methods  eg: c.ListServices?
Asking because I see this in a lot of places.